### PR TITLE
fix(cookieReader): safely access $document so it can be mocked

### LINKED ISSUE #11373: lastCookieString.split error with httpBackend.flush()
Using v1.4.0-build.3918+sha.4588e62 my unit test are now having the following error.

```
    TypeError: 'undefined' is not an object (evaluating 'lastCookieString.split'
)
        at C:/TR Forks/fef/bower_components/angular/angular.js:17354
        at sendReq (C:/TR Forks/fef/bower_components/angular/angular.js:9824)
        at C:/TR Forks/fef/bower_components/angular/angular.js:9542
        at processQueue (C:/TR Forks/fef/bower_components/angular/angular.js:140
60)
        at C:/TR Forks/fef/bower_components/angular/angular.js:14076
        at C:/TR Forks/fef/bower_components/angular/angular.js:15283
        at C:/TR Forks/fef/bower_components/angular/angular.js:15098
        at C:/TR Forks/fef/bower_components/angular-mocks/angular-mocks.js:1508
        at C:/TR Forks/fef/test/spec/services/FefLoagger.js:263
```

The unit test is this

```js
it('test mult call is correct format.', function() {
        var success = true;
        httpBackend.expectPOST('/demo/logs')
            .respond(
                function(method, url, data, headers) {
                    var validData = {
                        message: ['test'],
                        level: 'log',
                        name: 'fef:log1'
                    };
                    data = JSON.parse(data);
                    if (data.message[0] === validData.message[0] &&
                        data.level === validData.level &&
                        data.name === validData.name) {
                        return [200, {
                            token: '12345'
                        }];
                    } else {
                        success = false;
                        return [400, {
                            failure: 'bad'
                        }];
                    }
                });
        var log1 = FefLogger.getLogger('log1');
        log1.log('test');
        httpBackend.flush();
        expect(success).toBe(true);

    });
```

the line 263 coralates with the httpBackend.flush();
this unit test did work with 1.4.0-beta.0 and 1.3.15

### PR DESCRIPTION
**Pull request by shahata**
Closes #11373

### CODE REVIEW

**Comment by shahata** [comment]
I'm really not sure that this is the right way to solve this. I'm thinking it might be a good idea to have some mock implementation of `cookieReader`/`cookieWriter` in `angular-mocks` which can be configured during tests similarly to `$httpBackend`. We used to have something similar in `$browser`, but it wasn't documented and it got refactored out during the `ngCookies` re-write (which is why this problem only occurred now)

**Comment by Narretz** [comment]
@petebacondarwin probably comes up with a good idea how to solve this, he always does :)

File: src/ng/cookieReader.js [hunk 1]
```diff
@@ -22,6 +22,13 @@ function $$CookieReader($document) {
     }
   }
 
+  //null object implementation in case document was mocked out
+  if (!rawDocument || typeof rawDocument.cookie !== 'string') {
+    return function() {
+      return lastCookies;
+    };
+  }
+
```
Comment (petebacondarwin) [inline src/ng/cookieReader.js]: I think this is probably overkill. If the developer mocks out window and doesn't include a document then lots of stuff is going to break. Instead, how about we simply do

```js
lastCookieString = rawDocument.cookie || '';
```

In the function below.

Comment (shahata) [inline src/ng/cookieReader.js]: actually this is the only place that breaks, you can mock out `$window` and not include `$document`. See https://github.com/angular/angular.js/blob/master/test/ng/windowSpec.js

Comment (shahata) [inline src/ng/cookieReader.js]: I can also do something like this, if you prefer:
```js
rawDocument = $document[0] || {};
lastCookieString = rawDocument.cookie || '';
```

Comment (petebacondarwin) [inline src/ng/cookieReader.js]: Ugh! Mocking window without a document is surely crazy juice!
OK let's go with this version.

Comment (shahata) [inline src/ng/cookieReader.js]: okay, amended.

**Comment by petebacondarwin** [comment]
LGTM - feel free to merge

**Pull request closed without merging**